src/event.rs

```diff
@@ -23,6 +26,43 @@ use std::fmt::{self, Debug, Formatter};
 // FIXME - See https://maidsafe.atlassian.net/browse/MAID-2026 for info on removing this exclusion.
 #[allow(clippy::large_enum_variant)]
 pub enum Event {
+    // =========== Client Events - Routing won't handle these ===========
+    /// Inform the user (library) that we are connected to a client
+    ConnectedToClient {
+        /// Client's endpoint
+        peer_addr: SocketAddr,
+    },
+    /// Inform the user (library) that we are disconnected from a client
+    ConnectionFailureToClient {
+        /// Client's endpoint
+        peer_addr: SocketAddr,
+    },
+    /// Inform the user (library) that we have a new message from a client
+    NewMessageFromClient {
+        /// Client's endpoint
+        peer_addr: SocketAddr,
+        /// Client's message
+        msg: NetworkBytes,
+    },
+    /// Inform the user (library) that we couldn't send this message to a client
+    UnsentUserMsgToClient {
+        /// Client's endpoint
+        peer_addr: SocketAddr,
+        /// Message we had tried to send to the client
+        msg: NetworkBytes,
+        /// Token that we had used to identify this message
+        token: Token,
+    },
+    /// Inform the user (library) that we have successfully sent this message to a client
+    SentUserMsgToClient {
+        /// Client's endpoint
+        peer_addr: SocketAddr,
+        /// Message we had tried to send to the client
```

Comment:
If they have a logic that waits of success or failure i figured it could be useful. Since we have that for node-node and routing uses it, kinda unfair to not give that freedom to vault to client esp when it's coming from quic-p2p universally for all connections regardless.

I'm OK to remove it if you want though

Reply:
Let's leave, for now, it may be an optimisation.